tests/runtime/prelude/core/primitives/String.sk

```diff
@@ -248,194 +212,111 @@ native class .String uses Hashable, Show, Orderable {
   }
 
   fun trimLeft(): String {
-    offset = this.search(0, ch ~> !isWhitespace(ch));
-    if (offset.isSome()) {
-      this.substring(offset.fromSome())
-    } else {
-      ""
+    this.search(ch ~> !isWhitespace(ch)) match {
+    | Some(i) -> i.substring(this.getEndIter())
+    | None() -> ""
     }
   }
 
   fun trimRight(): String {
-    offset = this.rsearch(this.length() - 1, ch ~> !isWhitespace(ch));
-    if (offset.isSome()) {
-      this.sub(0, offset.fromSome() + 1)
-    } else {
-      ""
+    this.rsearch(ch ~> !isWhitespace(ch)) match {
+    | Some(i) ->
+      _ = i.next();
+      this.getIter().substring(i)
+    | None() -> ""
     }
   }
 
   fun trim(): String {
     this.trimLeft().trimRight()
   }
 
-  fun search(offset: Int, f: Char -> Bool): ?Int {
-    if (offset.uge(this.length())) {
-      None()
-    } else if (f(this.unsafe_get(offset))) {
-      Some(offset)
-    } else {
-      this.search(offset + 1, f)
-    }
-  }
-
-  fun searchIndex(offset: Int, f: Char -> Bool): Int {
-    if (offset.uge(this.length())) {
-      -1
-    } else if (f(this.unsafe_get(offset))) {
-      offset
-    } else {
-      this.searchIndex(offset + 1, f)
-    }
+  // Finds the given pattern in the string, starting from 'from'
+  fun search(f: Char -> Bool): ?mutable StringIterator {
```

Comment:
Ugh didn't realize we (still) had this method instead of renaming to "contains()".  Disregard my note about this method on StringIterator, we can rename both in a follow-up.